internal/webhooks/cluster.go

```diff
@@ -84,11 +84,37 @@ func (webhook *Cluster) Default(ctx context.Context, obj runtime.Object) error {
 			return apierrors.NewInternalError(errors.Wrapf(err, "Cluster %s can't be validated. ClusterClass %s can not be retrieved.", cluster.Name, cluster.Spec.Topology.Class))
 		}
 
-		defaultedVariables, allErrs := variables.DefaultClusterVariables(cluster.Spec.Topology.Variables, clusterClass.Spec.Variables, field.NewPath("spec", "topology", "variables"))
+		// We gather all defaulting errors and return them together.
+		var allErrs field.ErrorList
+
+		defaultedVariables, errs := variables.DefaultClusterVariables(cluster.Spec.Topology.Variables, clusterClass.Spec.Variables,
+			field.NewPath("spec", "topology", "variables"), true)
+		if len(errs) > 0 {
+			allErrs = append(allErrs, errs...)
+		} else {
+			cluster.Spec.Topology.Variables = defaultedVariables
+		}
+
+		if cluster.Spec.Topology.Workers != nil {
+			for i, md := range cluster.Spec.Topology.Workers.MachineDeployments {
+				// Continue if there are no variable overrides.
+				if md.Variables == nil || len(md.Variables.Overrides) == 0 {
```

Comment:
nit
```suggestion
				if len(md.Variables.Overrides) == 0 {
```

Reply:
This panics because `md.Variables` (the parent struct) is already nil.